addons/mail/static/src/web/messaging_service_patch.js

```diff
@@ -41,6 +42,26 @@ patch(Messaging.prototype, "mail/web", {
         }
         this._super(data);
     },
+    _handleNotificationRecordInsert(notif) {
+        this._super(notif);
+        if (notif.payload.Thread) {
+            const data = notif.payload.Thread;
+            const thread = this.store.threads[createLocalId(data.model, data.id)];
```

Comment:
This looks weak against race conditions? What if the thread was left beforehand?

Reply:
In earlier attempt I did `threadService.insert()`, but unfortunately there's some undesirable side-effects, e.g. it triggers some `onChange()` that make tests fail.

That's why I check presence here. I can add truthy `thread` in conditional to be more robust against leaving channel for example